compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -738,7 +738,16 @@ class ResolverImpl(
                     )
                     javaTypeParameterMap[typeParameterDescriptor] = psi
                 }
-                return getKSTypeCached(resolveJavaType(type.psi, type), type.element.typeArguments, type.annotations)
+                val resolved = getKSTypeCached(
+                    resolveJavaType(type.psi, type),
+                    type.element.typeArguments,
+                    type.annotations
+                )
+                return if (type.psi is PsiArrayType) {
+                    resolved
+                } else {
+                    resolved.replace(type.element.typeArguments)
+                }
```

Comment:
I'm a little bit confused. Why do you special-case Array here?

Also, I'm worried that replacing the type arguments in this way would effectively change the semantics of platform types.